nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -498,4 +553,92 @@ public void compareGenericTypeParameterNullabilityForCall(
       }
     }
   }
+
+  public void checkTypeParameterNullnessForMethodOverriding(
+      MethodTree tree, Symbol.MethodSymbol overridingMethod, Symbol.MethodSymbol overriddenMethod) {
+    if (!config.isJSpecifyMode()) {
+      return;
+    }
+    checkTypeParameterNullnessForOverridingMethodReturnType(tree, overridingMethod);
+    checkTypeParameterNullnessForOverridingMethodParameterType(
+        tree, overridingMethod, overriddenMethod);
+  }
+
+  public Nullness getMethodParamNullness(
+      int paramIndex, Symbol.MethodSymbol methodSymbol, Tree tree) {
+    if (tree instanceof JCTree.JCMethodInvocation == false) {
+      return Nullness.NONNULL;
+    }
+    JCTree.JCMethodInvocation methodInvocationTree = (JCTree.JCMethodInvocation) tree;
+    if (methodInvocationTree.meth instanceof JCTree.JCFieldAccess == false) {
+      return Nullness.NONNULL;
+    }
+    Type methodReceiverType = ((JCTree.JCFieldAccess) methodInvocationTree.meth).selected.type;
+    Type methodType = state.getTypes().memberType(methodReceiverType, methodSymbol);
+    Type formalParamType = methodType.getParameterTypes().get(paramIndex);
+    boolean hasNullableAnnotation =
+        Nullness.hasNullableAnnotation(formalParamType.getAnnotationMirrors().stream(), config);
+    if (hasNullableAnnotation) {
+      return Nullness.NULLABLE;
+    } else {
+      return Nullness.NONNULL;
+    }
+  }
+
+  private void checkTypeParameterNullnessForOverridingMethodParameterType(
+      MethodTree tree, Symbol.MethodSymbol overridingMethod, Symbol.MethodSymbol overriddenMethod) {
+    List<? extends VariableTree> methodParameters = tree.getParameters();
+    Type methodWithTypeParams =
+        state.getTypes().memberType(overridingMethod.owner.type, overriddenMethod);
+    List<Type> typeParameterTypes = methodWithTypeParams.getParameterTypes();
+    for (int i = 0; i < methodParameters.size(); i++) {
+      Type methodParameterType = ASTHelpers.getType(methodParameters.get(i));
+      Type typeParameterType = typeParameterTypes.get(i);
+      if (typeParameterType instanceof Type.ClassType
+          && methodParameterType instanceof Type.ClassType) {
+        boolean hasNullableAnnotationTypeParamType =
+            Nullness.hasNullableAnnotation(
+                typeParameterType.getAnnotationMirrors().stream(), config);
+        boolean hasNullableAnnotationMethodParameterType =
+            Nullness.hasNullableAnnotation(
+                methodParameterType.getAnnotationMirrors().stream(), config);
+        if (hasNullableAnnotationTypeParamType != hasNullableAnnotationMethodParameterType) {
+          reportInvalidOverridingMethodParamTypeError(
+              methodParameters.get(i), typeParameterType, methodParameterType);
+        }
```

Comment:
This is the wrong check, as it forces the nullability of the parameter in the overriding method to be the same as the overridden method.  But, if the overridden method has a `@NonNull` parameter, it is allowed for the overriding method to have a `@Nullable` parameter.  I have added a corresponding test case.

This code here should be deleted.  Instead, around line 728 in `NullAway.java`, we should add another call to `getMethodParamNullness()` so that `overriddenMethodArgNullnessMap[i]` is set correctly for code using generics in JSpecify mode.  We do want to keep the check below that the type param nullability annotations match.

Reply:
Thanks for explaining. I was trying to compare the nullability annotations of the type parameter and the method parameter of the overriding method here but realized it is the wrong place to do so. I updated the code as per your suggestion.